### PR TITLE
feat(murmur): add murmuring config, nudge tracker, whisper budget, and list command

### PR DESCRIPTION
**Pull request by rsnodgrass**
## Summary

- Adds `murmuring` project config field (defaults to "off") that enables periodic nudging of AI coworkers to self-report what they're working on via `ox murmur`
- Introduces `MurmurNudgeTracker` in the daemon to track agent heartbeats and trigger nudges after configurable intervals, with early-nudge logic for new agents
- Adds `capMurmurWhispers` to enforce a token budget on murmur whispers delivered to agents — per-agent cap with random sampling for fairness across long sessions
- Reduces max murmur content from 4096 to 500 bytes to keep coordination signals concise
- Adds `ox murmur list` subcommand with table/JSON output, filtering by topic, scope, agent, and time window
- Adds doctor checks registry (`internal/doctor/checks/registry.go`) for centralized check construction
- Comprehensive test coverage for all new components

## Test plan

- [x] `go build ./cmd/ox/` compiles cleanly
- [x] `go test ./internal/daemon/ -run TestMurmurNudgeTracker` passes (including cleanup regression test)
- [ ] `make test` full suite
- [ ] `make lint` passes
- [ ] Manual: `ox murmur list` with/without murmurs in ledger

Co-Authored-By: [SageOx](https://github.com/SageOx)

## Summary by CodeRabbit

* **New Features**
  * Added "murmur list" and "murmur delete" commands; list supports windowing, filtering, JSON or table output; delete commits removals.
  * Automatic murmur nudging to prompt inactive agents.

* **Changes**
  * Project-level murmuring mode (manual | auto) with caching and configurable nudge interval (default 15m).
  * Murmur content size limit reduced to 500 bytes.
  * Murmur inclusion now enforces 24‑hour window, per‑agent and total budgeted caps with fairness sampling.

* **Tests**
  * Extensive unit and integration tests covering listing, capping, nudging, config, and edge cases.

### CODE REVIEW

File: cmd/ox/murmur_list.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+	// ledger murmurs
+	if scopeFilter == "" || scopeFilter == "ledger" {
+		ledgerPath := getLedgerPath()
+		if ledgerPath != "" {
+			murmurs, readErr := ledger.ReadMurmursInWindow(ledgerPath, windowHours)
+			if readErr == nil {
+				allMurmurs = append(allMurmurs, murmurs...)
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/murmur_list.go]: Not applicable. `config.DefaultLedgerPath(repoID, endpointURL)` requires parameters not available in this CLI context. `getLedgerPath()` handles the full resolution chain (local config → default fallback) and is the same pattern used in `murmur.go:172` and `doctor_ledger_git.go`.

File: internal/daemon/sync.go [hunk 2]
```diff
@@ -901,6 +901,14 @@ func (s *SyncScheduler) doPull(ctx context.Context, progress *ProgressWriter, fo
 	s.ledgerMu.Lock()
 	result := func() ManagedRepoPullResult {
 		defer s.ledgerMu.Unlock()
+
+		// refresh sparse checkout so the rolling murmur window stays current.
+		// Without this, hourly directories computed at clone time go stale and
+		// new murmur files pulled from remote are not materialized on disk.
+		if err := ledger.ConfigureSparseCheckout(s.config.LedgerPath); err != nil {
+			s.logger.Warn("failed to refresh ledger sparse checkout", "error", err)
+		}
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply internal/daemon/sync.go]: I’m not fixing this. The current behavior is correct. Here’s why:

A failed sparse checkout refresh only affects murmur directory visibility — it does NOT corrupt the repo or break the pull. Aborting the entire pull would be worse because it blocks all ledger syncing (sessions, github data, audit) just because murmur paths couldn’t be updated.

The failure mode of the current code: stale murmur window (degraded, murmurs not visible until next successful refresh).

The failure mode of the proposed change: no ledger sync at all (sessions stop uploading, github data stops syncing, everything breaks).

Warn is the right log level — it’s a degraded state that self-heals on the next tick, not a fatal condition. Promoting to Error and aborting would turn a minor murmur visibility issue into a full sync outage.
